MidiPlayer/src/main/java/org/primesoft/midiplayer/track/BasePlayerTrack.java

```diff
@@ -50,12 +53,12 @@
  */
 public abstract class BasePlayerTrack extends BaseTrack {
 
-    private final HashSet<Player> m_players;
+    private final Set<Player> m_players;
 
     @Override
-    protected Player[] getPlayers() {
+    protected Collection<? extends Player> getPlayers() {
         synchronized (m_players) {
-            return m_players.toArray(new Player[0]);
+            return m_players;
```

Comment:
This part needs to create a new instance of a collection. Otherwise it might cause concurrent modification errors.

Reply:
Now returning new HashSet<>(m_players)